### PR TITLE
DOPS-3309: Add sonar & defectdojo analysis

### PR DESCRIPTION
**Pull request by BAStos525**
DOPS-3309: Add sonar & defectdojo analysis

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -105,3 +108,71 @@ task allShadowJars {
...
+      property "sonar.projectName", "${project.group}:${rootProject.name}.${project.name}"
+      property "sonar.sources", "${project.projectDir}/src/main/java"
+      property "sonar.tests", "${project.projectDir}/src/test"
+      property "sonar.java.test.binaries", "${project.projectDir}/build/test-results/test/binary"
+      property "sonar.junit.reportPaths", "${project.projectDir}/build/test-results/test/"
+      property "sonar.coverage.jacoco.xmlReportPaths", "${project.projectDir}/build/reports/testCoverage/*.xml"
+    }
+}
+
+// tasks.named('testCoverage') {
```
Comment (Mingela) [inline build.gradle]: why are these commented?

**Review by Mingela — APPROVED** [review]

File: .github/workflows/iroha2-pr.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 name: Iroha2-java pull requests workflow
 
 on:
-  pull_request:
-    branches: [ iroha2-dev, iroha2-main ]
+  push:
+    branches: [feature/DOPS-3309/sonar-dojo]
```
Comment (Mingela) [inline .github/workflows/iroha2-pr.yml]: should be fixed

**Review by arndey — APPROVED** [review]

File: .github/workflows/iroha2.yml [hunk 3]
```diff
@@ -23,8 +23,26 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
-      - name: Build with Gradle
-        run: ./gradlew build
+      - name: Build with Gradle & Sonarqube analysis
+        run: |
+          ./gradlew build
+          ./gradlew jacocoTestReport
```
Comment (Mingela) [inline .github/workflows/iroha2.yml:29]: is `test` executed implicitly?

Comment (BAStos525) [reply .github/workflows/iroha2.yml:29]: `gradlew tests` is included into `gradlew build` @arndey

**Pull request merged**